### ISSUE TITLE
Radford+, "Improving Language Understanding by Generative Pre-training", 2018

### ISSUE THREAD

**Issue by k-ishiguro** [k-ishiguro/Papers-metome#9]
GPT-1です。温故知新というか、chatGPTへの流れをきちんと把握しておきたくて読みました。
今読むと、本当にわかりやすく書けている論文だと思います。クレームが明確なのがいいのかも。

## 論文書誌情報

**第一著者 and/or 有名人著者**
Alec Radford, Ilya Sutskever (OpenAI)

**発表場所と発表年**
2018

**paper link**
https://openai.com/blog/language-unsupervised/

### この研究のクレームは何か？

大量のunlabled text dataを上手に様々なsupervised taskにTarnsferしたいという共通の課題に対して、pre-training + fine-tuneする課題。これに対して、
大量の文章全体を丸ごとTransformerに突っ込んで、n-gram尤度最大化だけの言語モデルをTransformerでpre-trainingするのと、入力側を工夫することで種々のNLPタスクにfinetuneするやりかたで、多数のタスクでSotAを更新できる。

### 先行研究とくらべて何がすごい？

(i) ファインチューンするときのネットワークのいじりを最小限にするための、文書入力側の設計方法の提案
(ii) word-embeddingによるPre-train transferは既に当時でもMNTなので強さを発揮していたが、明らかに単語共起意味だけしかキャプチャーできない。Transformerによるsentence -> latent sequenceをまるごと学習して、まるごとTransferするともっといろいろキャプチャーしているはず

これまでのTransferは、埋め込みの最後、識別の直前に帽子をかぶせてなんとかするのが多かったけど、これはタスクごとに入力形式の違いを埋められないので大変だった、という結構実務的につらい課題にsolutionを提案した。

### 技術や手法の強さはどこにあるか？

Tranformerのキャパシティが無茶苦茶でかいのはわかっていて、それを手軽かつ大体定まった方法で色々なタスクにfine-tuneできるようにした、入力の設計方法の提案。

Transfomer自体は、シーケンス一個を受けてk-gram尤度を最適化する言語モデル。

基本的に、固まり入力をdelimiterでつなげて一つの文としてTransformerに突っ込む。
多肢入力の場合は、それぞれ入力して、最後にFC Linear/Softmax. 

とにかく、transformer部分をいじらず全部使えるのが大事

#### そのほか実装・技術面のディテール

fine-tuneの目的関数は、言語モデルも入れておいた方が良さそうというのが実験からわかった
